File: NOVEMBER-2025/16-11-2025/1513.c
#define MOD 1000000007;
int numSub(char* s) {
    long int count = 0;
    long int current = 0;
    int len = strlen(s);

    for(int i = 0; i < len; i++){
        if(s[i] == '1'){
            current++;
            count = (count + current) % MOD;
        }
        else{
            current = 0;
        }
    }

    return count;
}

File: NOVEMBER-2025/16-11-2025/1513.dart
class Solution {
  int numSub(String s) {
    int count=0;
    int result=0;
    int sumofn=0;
    for(int i=0;i<s.length;i++){
        if(s[i]=='1'){
            count++;
        }else{
            sumofn=(count*(count+1))~/2;
            result+=sumofn;
            count=0;
        }
    }
    sumofn=(count*(count+1))~/2;
    result+=sumofn;
    return result%1000000007;
    }
}